### PR TITLE
Add support for ChannelData field

### LINKED ISSUE #1274: Add support for channelData field in conversation references
## Describe the bug

The Javascript SDK required an update to the ConversationReference class and createConversation method in order to include the channelData information, which is required by MS Teams.

See [botbuilder-js/685](https://github.com/Microsoft/botbuilder-js/issues/685) for the original bug, and [this pull request](https://github.com/Microsoft/botbuilder-js/pull/712) for the solution.

## Expected behavior

* Conversation references will include a channelData field.
* When calling createConversation, the channelData field will be populated with the tenant information if specified in the reference

### PR DESCRIPTION
**Pull request by JuanAr**
### Description
- Conversation references will include a channelData field.
- When calling createConversation, the channelData field will be populated with the tenant information if specified in the reference

### Changes made
- `CreateConversationAsync` now has an overload that takes a `ConversationReference` as a parameter

- `ConversationReference` now has ChannelData as a field, with the type `object`

- If the value is given `ChannelData` gets populated by an object with a TenantId `{tenant: tenantId }`

This PR fixes #1274
### To test it
* Connect a bot to Teams. Follow the steps on [this documentation](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/bots/bots-test)

* Using a `ConversationReference` extracted from an incoming _group_ message, use `adapter.createConversationAsync` to create a new _private_ context.

* Use the _new context_ to send a private message to the user.  Note that it is received as a **DM**, instead of a **reply** in the group channel.

### CODE REVIEW

**Comment by JuanAr** [comment]
@benbrown we created this PR following the same logic in Microsoft/botbuilder-js#712.

However, while we were able to partially test it in Teams using [16.proactive-messages](https://github.com/Microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/16.proactive-messages) it will be helpful to have a second opinion on the solution and the testing since the sample uses a different method for proactive messages than JS.

**Comment by benbrown** [comment]
Great! I'll take a look asap.  There are some small details that are shifting as we speak with regards to the final location of the tenant id field. 

Stay tuned!

File: libraries/Microsoft.Bot.Schema/ConversationReference.cs [hunk 1]
```diff
@@ -38,14 +38,15 @@ public ConversationReference()
...
-        public ConversationReference(string activityId = default(string), ChannelAccount user = default(ChannelAccount), ChannelAccount bot = default(ChannelAccount), ConversationAccount conversation = default(ConversationAccount), string channelId = default(string), string serviceUrl = default(string))
+        public ConversationReference(string activityId = default(string), ChannelAccount user = default(ChannelAccount), ChannelAccount bot = default(ChannelAccount), ConversationAccount conversation = default(ConversationAccount), string channelId = default(string), string serviceUrl = default(string), object channelData = default(object))
         {
             ActivityId = activityId;
             User = user;
             Bot = bot;
             Conversation = conversation;
             ChannelId = channelId;
             ServiceUrl = serviceUrl;
+            ChannelData = channelData;
```
Comment (benbrown) [inline libraries/Microsoft.Bot.Schema/ConversationReference.cs]: no longer necessary. the required field wil be in the Conversation object.

File: libraries/Microsoft.Bot.Schema/ConversationReference.cs [hunk 2]
```diff
@@ -89,7 +90,12 @@ public ConversationReference(string activityId = default(string), ChannelAccount
         /// referenced conversation may be performed
         /// </summary>
         [JsonProperty(PropertyName = "serviceUrl")]
-        public string ServiceUrl { get; set; }
-
+        public string ServiceUrl { get; set; }    
+        
+        /// <summary>
```
Comment (benbrown) [inline libraries/Microsoft.Bot.Schema/ConversationReference.cs]: these files are autogenerated based on the swagger definition, so any changes here should be handled automatically.

**Comment by cleemullins** [comment]
@benbrown Let's sync on this tomorrow and make sure we've got a plan in place.

**Comment by benbrown** [comment]
I have updated the Javascript version of this change to reflect the current state of things.

There is a short term fix and a longer term one, because changes have to be made down to the Teams-API level.  There is also a schema change happening in the swagger files, which will have to be rebuilt for this to work.  The changes to swagger are underway now.

**Comment by JuanAr** [comment]
@benbrown we just reverted the schema changes and left the short term workaround in place. We haven't included the long term solution since we are not updating the `ConversationReference` class.

**Comment by benbrown** [comment]
@JuanAr I believe we have the final set of changes in place here:

https://github.com/Microsoft/botbuilder-js/pull/712/files

* add tenantId to ConversationParameters
* add tenantId to ConversationAccount

* Make sure we are setting tenantId as part of a call to createConversation if present in reference.conversation.tenantId
* Set it in both new and old places (parameters.tenantId and parameters.channelData.tenant.id)

**Comment by JuanAr** [comment]
@benbrown we applied the latest changes. Also, update the swagger file to update the model. Let me know if there's some additional changes.

File: libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs [hunk 3]
```diff
@@ -778,19 +778,22 @@ public virtual async Task CreateConversationAsync(string channelId, string servi
         /// </remarks>
         public virtual async Task CreateConversationAsync(string channelId, string serviceUrl, MicrosoftAppCredentials credentials, ConversationParameters conversationParameters, BotCallbackHandler callback, ConversationReference reference, CancellationToken cancellationToken)
         {
-            if (reference.ChannelData != null)
+            if (reference.Conversation != null)
             {
-                var typeOfDynamic = reference.ChannelData.GetType();
+                var typeOfDynamic = reference.Conversation.GetType();
                 var tenantProperty = typeOfDynamic.GetProperty("tenant");
```
[the remark this answers is not in the document]
Comment (benbrown) [reply libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs]: I think this still needs to be updated!  Other than that this is good to go.

**Comment by benbrown** [comment]
@JuanAr - there is one more minor change needed I think. See comments above -- apologies if these weren't visible before I think Github hit them because I hadn't officially submitted the review.

**Pull request merged**